File: giorno_5 es3.py
# giorno 5 es 3
def trova_valore_massimo(arr):
  if not arr:
    return None

  val_max = arr[0]
  for elemento in arr:
    if elemento > val_max:
      val_max = elemento
  return val_max

my_array = [25, 8, 42, 13]
valore_massimo = trova_valore_massimo(my_array)

if valore_massimo is not None:
  print(f"Il valore massimo nell'array è: {valore_massimo}")
else:
  print("L'array è vuoto.")

File: giorno_5 es3.js
function trovaValoreMassimo(arr) {
  if (arr.length === 0) {
    return null;
  }

  let valMax = arr[0];
  for (let i = 1; i < arr.length; i++) {
    if (arr[i] > valMax) {
      valMax = arr[i];
    }
  }
  return valMax;
}

const my_array = [25, 8, 42, 13];
const valoreMassimo = trovaValoreMassimo(my_array);

if (valoreMassimo !== null) {
  console.log(`Il valore massimo nell'array è: ${valoreMassimo}`);
} else {
  console.log("L'array è vuoto.");
}